tests/Source/DoctrineODMQuerySourceIteratorTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Exporter\Tests\Source;
+
+use Doctrine\ODM\MongoDB\Configuration;
+use Doctrine\ODM\MongoDB\DocumentManager;
+use PHPUnit\Framework\TestCase;
+use Sonata\Exporter\Source\DoctrineODMQuerySourceIterator;
+use Sonata\Exporter\Tests\Source\Fixtures\Document;
+
+final class DoctrineODMQuerySourceIteratorTest extends TestCase
+{
+    /** @var DocumentManager */
+    private $dm;
+
+    protected function setUp(): void
+    {
+        if (!\extension_loaded('mongodb')) {
+            $this->markTestSkipped('The mongodb extension is not available.');
+        }
+
+        $this->dm = DocumentManager::create(null, $this->createConfiguration());
+
+        $documentA = new Document();
+        $documentB = new Document();
+
+        $this->dm->persist($documentA);
+        $this->dm->persist($documentB);
+        $this->dm->flush();
+    }
+
+    protected function tearDown(): void
+    {
+        $this->dm->createQueryBuilder(Document::class)
+            ->remove()
+            ->getQuery()
+            ->execute();
+    }
+
+    public function testHandler(): void
+    {
+        $query = $this->dm->createQueryBuilder(Document::class)->getQuery();
+
+        $iterator = new DoctrineODMQuerySourceIterator($query, ['id']);
+
+        $results = [];
```

Comment:
Why not [`iterator_to_array()`](https://www.php.net/manual/en/function.iterator-to-array.php)?

Reply:
Because laziness, I had copied it from https://github.com/sonata-project/exporter/blob/2.x/tests/Source/PDOStatementSourceIteratorTest.php#L68 and didn't look too much 😬, changed.